src/agentscope/agents/rag_agents.py

```diff
@@ -0,0 +1,339 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build an agent with RAG
+(with LlamaIndex or Langchain)
+"""
+
+from abc import ABC, abstractmethod
+from typing import Optional, Any
+import importlib
+from loguru import logger
+
+
+from agentscope.agents.agent import AgentBase
+from agentscope.message import Msg
+from agentscope.prompt import PromptEngine
+from agentscope.models import load_model_by_config_name
+
+try:
+    from agentscope.rag import RAGBase, LlamaIndexRAG
+except ImportError as exc:
+    raise ImportError(
+        "Need to install llama index for RAG agents.",
+    ) from exc
+
+
+class RAGAgentBase(AgentBase, ABC):
+    """
+    Base class for RAG agents
+    """
+
+    def __init__(
+        self,
+        name: str,
+        sys_prompt: str,
+        model_config_name: str,
+        emb_model_config_name: str,
+        memory_config: Optional[dict] = None,
+        rag_config: Optional[dict] = None,
+    ) -> None:
+        """
+        Initialize the RAG base agent
+        Args:
+            name (str):
+                the name for the agent.
+            sys_prompt (str):
+                system prompt for the RAG agent.
+            model_config_name (str):
+                language model for the agent.
+            emb_model_config_name (str):
+                embedding model for the agent.
+            memory_config (dict):
+                memory configuration.
+            rag_config (dict):
+                config for RAG. It contains most of the
+                important parameters for RAG modules. If not provided,
+                the default setting will be used.
+                Examples can refer to children classes.
+        """
+        super().__init__(
+            name=name,
+            sys_prompt=sys_prompt,
+            model_config_name=model_config_name,
+            use_memory=True,
+            memory_config=memory_config,
+        )
+        # init prompt engine
+        self.engine = PromptEngine(self.model)
```

Comment:
please use `model.format` instead and remove `self.engine`

Reply:
updated.